### PR TITLE
Flat quick chat

### PR DESCRIPTION
**Pull request by dtracers**
Quick chat is now done through flat buffers.

Callbacks are still not implemented correctly

### CODE REVIEW

File: src/main/cpp/RLBotInterface/RLBotInterface/GameFunctions/PlayerInfo.cpp [hunk 1]
```diff
@@ -20,6 +20,15 @@ namespace GameFunctions
 		return RLBotCoreStatus::Success;
 	}
 
+	// FLAT
+	static BoostUtilities::QueueSender quickChatQueue(BoostConstants::QuickChatFlatQueueName);
+
+	extern "C" RLBotCoreStatus RLBOT_CORE_API SendQuickChat(void* quickChatMessage, int protoSize)
+	{
+		DEBUG_LOG("CHATTING");
```
Comment (tarehart) [inline src/main/cpp/RLBotInterface/RLBotInterface/GameFunctions/PlayerInfo.cpp]: Can probably remove this now

File: src/main/cpp/RLBotInterface/RLBotInterface/GameFunctions/PlayerInfo.hpp [hunk 2]
```diff
@@ -17,6 +17,7 @@ extern "C"
 	namespace GameFunctions
 	{
 		RLBotCoreStatus checkQuickChatPreset(QuickChatPreset quickChatPreset);
+		DLL_EXPORT RLBotCoreStatus RLBOT_CORE_API SendQuickChat(void* controllerState, int protoSize);
```
Comment (tarehart) [inline src/main/cpp/RLBotInterface/RLBotInterface/GameFunctions/PlayerInfo.hpp]: param is still named controllerState

File: src/main/python/rlbot/agents/base_agent.py [hunk 3]
```diff
@@ -48,11 +48,13 @@ def get_output_vector(self, game_tick_packet):
 
     def send_quick_chat(self, team_only, quick_chat):
         """
-        Sends a quick chat to the other bots.
+        Sends a quick chat to the other bots.  If it is None it does not send a quick chat to other bots
```
Comment (tarehart) [inline src/main/python/rlbot/agents/base_agent.py]: This new sentence is a bit confusing. Are you talking about QuickChats.CHAT_NONE or python's None? Why would somebody pass either of those?

File: src/main/python/rlbot/agents/base_flatbuffer_agent.py [hunk 4]
```diff
@@ -41,9 +41,9 @@ def get_output_flatbuffer(self, game_tick_flatbuffer):
         ControllerState.ControllerStateAddPitch(builder, friendly_state.pitch)
         ControllerState.ControllerStateAddYaw(builder, friendly_state.yaw)
         ControllerState.ControllerStateAddRoll(builder, friendly_state.roll)
-        ControllerState.ControllerStateAddJump(builder, friendly_state.jump)
-        ControllerState.ControllerStateAddBoost(builder, friendly_state.boost)
-        ControllerState.ControllerStateAddHandbrake(builder, friendly_state.handbrake)
+        ControllerState.ControllerStateAddJump(builder, int(friendly_state.jump))
```
Comment (tarehart) [inline src/main/python/rlbot/agents/base_flatbuffer_agent.py]: What's the theory here?

Comment (dtracers) [reply src/main/python/rlbot/agents/base_flatbuffer_agent.py]: Removed

File: src/main/cpp/RLBotInterface/RLBotInterface/GameFunctions/PlayerInfo.cpp (hunk 1 above)
Comment (dtracers) [reply src/main/cpp/RLBotInterface/RLBotInterface/GameFunctions/PlayerInfo.cpp]: Done

File: src/main/cpp/RLBotInterface/RLBotInterface/GameFunctions/PlayerInfo.hpp (hunk 2 above)
Comment (dtracers) [reply src/main/cpp/RLBotInterface/RLBotInterface/GameFunctions/PlayerInfo.hpp]: Done

File: src/main/python/rlbot/agents/base_agent.py (hunk 3 above)
Comment (dtracers) [reply src/main/python/rlbot/agents/base_agent.py]: This will probably mostly be used for debugging purposes.

**Pull request merged**